controller/state.go

```diff
@@ -777,15 +786,17 @@ func (m *appStateManager) CompareAppState(app *v1alpha1.Application, project *v1
 			// the source object, don't store sync status, and do not affect
 			// overall sync status
 		} else if !isManagedNs && (diffResult.Modified || targetObj == nil || liveObj == nil) {
-			// Set resource state to OutOfSync since one of the following is true:
-			// * target and live resource are different
-			// * target resource not defined and live resource is extra
-			// * target resource present but live resource is missing
-			resState.Status = v1alpha1.SyncStatusCodeOutOfSync
-			// we ignore the status if the obj needs pruning AND we have the annotation
-			needsPruning := targetObj == nil && liveObj != nil
-			if !(needsPruning && resourceutil.HasAnnotationOption(obj, common.AnnotationCompareOptions, "IgnoreExtraneous")) {
+			// set the overall AND resource sync status to OutOfSync,
+			// unless the resource requires pruning and has the IgnoreExtraneous annotation
+			if resState.RequiresPruning && resState.IgnoreExtraneous {
+				resState.Status = v1alpha1.SyncStatusCodeSynced
```

Comment:
I think resource should still show out of sync to be clear, the app will show as in sync. Maybe we can have a separate state like yellow heart for not-synced-but-ignored.

Reply:
@andrii-korotkov-verkada the reason I made the status "Synced" when we are ignoring it is because of the 3 options, it seemed like the most reflective of the situation.

I don't think we should add an "ignored" sync status, because that's really not a sync status, and just hides the underlying information about if we would make a change when syncing.

It's also more helpful in the UI, because users likely don't want to see ignored resource when they filter by "OutOfSync" in the UI.

---

I think we need to agree what sync-status is representing, is it:

1. an indication of if the app/resource is in the state we "want" it in
2. an indication of if we will make some change when the resource is synced

If it's `1`, then I think `IgnoreExtraneous` is a clear indication that the user does not care if this resource needs to be pruned, so the PRs current behavior makes sense. (Remembering that the resource will still be pruned unless `Prune=false`, although it's hard to think of situations where you would not set both).

If it's `2`, then I think the addition of `IgnoreExtraneous` was a mistake, because it decoupled the concepts. However, because it alraedy exists, I guess we should set the resource to Synced if BOTH `IgnoreExtraneous` and `Prune=false`, otherwise it should be "OutOfSync".

Interested to know your thoughts?